tests/py_test.py

```diff
@@ -83,6 +84,449 @@ def test_wire_format(self):
         f.close()
 
 
+class TestObjectBasedAPI(unittest.TestCase):
```

Comment:
I like the tests you've added, nice work! WDYT about making a function like `CheckReadBuffer` but for objects? Then you can use it to test more situations.

Reply:
Thanks Robert for your comments! The tests of the object API are actually using CheckReadBuffer. So it tests if the buffer converted by the object API can pass CheckReadBuffer test. This is kind of a work around way to test the object API, but it has an advantage to keep the consistency between the object API and the buffer API.